### PR TITLE
sequent 8 use tty prompt

### PR DESCRIPTION
**Pull request by lvonk**
- **Use gli and tty-prompt to improve sequent cli**

### CODE REVIEW

File: lib/sequent/cli/sequent8_migration.rb [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# frozen_string_literal: true
```
Comment (erikrozendaal) [inline lib/sequent/cli/sequent8_migration.rb:1]: Rename this file to `sequent_8_migration.rb` for consistency?

File: lib/sequent/cli/sequent8_migration.rb [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+            ```
+            psql> ROLLBACK;
+            ```
+          EOS
+        end
+        answer
+      end
+
+      def sequent_gem_dir = Gem::Specification.find_by_name('sequent').gem_dir
```
Comment (erikrozendaal) [inline lib/sequent/cli/sequent8_migration.rb:172]: This breaks when running directly from the sequent source directory. Is there a way to make a gem work without having to install it first? Or another easy way to test this code without having to install every time?

File: sequent.gemspec [hunk 3]
```diff
@@ -39,14 +40,15 @@ Gem::Specification.new do |s|
...
+  s.add_dependency 'tty-prompt', '~> 0.23.1'
   s.add_dependency 'tzinfo', '>= 1.1'
   s.add_development_dependency 'prop_check', '~> 1.0'
   s.add_development_dependency 'pry', '~> 0.13'
   s.add_development_dependency 'rake', '~> 13'
   s.add_development_dependency 'rspec', rspec_version
   s.add_development_dependency 'rspec-collection_matchers', '~> 1.2'
   s.add_development_dependency 'rspec-mocks', rspec_version
-  s.add_development_dependency 'rubocop', '~> 1.56', '>= 1.56.3'
+  s.add_development_dependency 'rubocop', '~> 1.68.0'
```
Comment (erikrozendaal) [inline sequent.gemspec:51]: Also update .rubocop.yml ? Or add:

```yaml
  AllCops:
    NewCops: enable
```

**Review by erikrozendaal — APPROVED** [review]
Nice!

File: lib/sequent/cli/sequent8_migration.rb (hunk 2 above)
Comment (lvonk) [reply lib/sequent/cli/sequent8_migration.rb:172]: Yes, you could probably do something like:

`some_file.rb`:
```ruby
require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  gem "sequent", git: 'https://github.com/zilverline/sequent'
end

require 'gli'
require 'tty-prompt'
require './lib/version'
require './lib/sequent/cli/app'

exit_code = Sequent::Cli::App.run(ARGV)
exit(exit_code)
```

File: lib/sequent/core/helpers/attribute_support.rb [hunk 4]
```diff
@@ -63,12 +63,11 @@ def attrs(args)
...
-                associations << attribute
-              elsif included_modules.include?(ActiveModel::Validations) &&
-                    type.included_modules.include?(Sequent::Core::Helpers::AttributeSupport)
-                associations << attribute
-              end
+              next unless type.instance_of?(Sequent::Core::Helpers::ArrayWithType) ||
+                          (included_modules.include?(ActiveModel::Validations) &&
+                              type.included_modules.include?(Sequent::Core::Helpers::AttributeSupport))
+
+              associations << attribute
```
Comment (erikrozendaal) [inline lib/sequent/core/helpers/attribute_support.rb:70]: Not sure if this is better, `unless` already breaks my brain and then we have this complicated condition...

**Review by erikrozendaal — APPROVED** [review]

File: lib/sequent/core/helpers/attribute_support.rb (hunk 4 above)
Comment (lvonk) [reply lib/sequent/core/helpers/attribute_support.rb:70]: I agree, it was rubocop's suggestion...

**Review by erikrozendaal — APPROVED** [review]

**Pull request merged**